contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,568 @@
+
+    function initialize(
+        address _xvsVault,
+        address _xvsVaultRewardToken,
+        uint256 _xvsVaultPoolId,
+        uint128 _alphaNumerator,
+        uint128 _alphaDenominator,
+        address _accessControlManager,
+        address _protocolShareReserve,
+        address _comptroller
+    ) external virtual initializer {
+        alphaNumerator = _alphaNumerator;
+        alphaDenominator = _alphaDenominator;
+        xvsVaultRewardToken = _xvsVaultRewardToken;
+        xvsVaultPoolId = _xvsVaultPoolId;
+        xvsVault = _xvsVault;
+        nextScoreUpdateRoundId = 0;
+        protocolShareReserve = _protocolShareReserve;
+        comptroller = _comptroller;
+
+        __AccessControlled_init(_accessControlManager);
+    }
+
+    /**
+     * @notice Update value of alpha
+     * @param _alphaNumerator numerator of alpha. If alpha is 0.5 then numberator is 1
+     * @param _alphaDenominator denominator of alpha. If alpha is 0.5 then numberator is 2
+     */
+    function updateAlpha(uint128 _alphaNumerator, uint128 _alphaDenominator) external {
+        _checkAccessAllowed("updateAlpha(uint128,uint128)");
+
+        alphaNumerator = _alphaNumerator;
+        alphaDenominator = _alphaDenominator;
+
+        for (uint i = 0; i < allMarkets.length; i++) {
+            accrueInterest(allMarkets[i]);
+        }
+
+        _startScoreUpdateRound();
+    }
+
+    /**
+     * @notice Update multipliers for a market
+     * @param _supplyMultiplier new supply multiplier for the market
+     * @param _borrowMultiplier new borrow multiplier for the market
+     */
+    function updateMultipliers(address market, uint256 _supplyMultiplier, uint256 _borrowMultiplier) external {
+        _checkAccessAllowed("updateMultipliers(address,uint256,uint256)");
+        require(markets[market].exists == true, "market is not supported");
+
+        accrueInterest(market);
+        markets[market].supplyMultiplier = _supplyMultiplier;
+        markets[market].borrowMultiplier = _borrowMultiplier;
+
+        _startScoreUpdateRound();
+    }
+
+    /**
+     * @notice Add a market to prime program
+     * @param vToken address of the market vToken
+     * @param supplyMultiplier the multiplier for supply cap. It should be converted to 1e18
+     * @param borrowMultiplier the multiplier for borrow cap. It should be converted to 1e18
+     */
+    function addMarket(address vToken, uint256 supplyMultiplier, uint256 borrowMultiplier) external {
+        _checkAccessAllowed("addMarket(address,uint256,uint256)");
+        require(markets[vToken].exists == false, "market is already added");
+
+        markets[vToken].rewardIndex = 0;
+        markets[vToken].supplyMultiplier = supplyMultiplier;
+        markets[vToken].borrowMultiplier = borrowMultiplier;
+        markets[vToken].score = 0;
+        markets[vToken].exists = true;
+
+        vTokenForAsset[IVToken(vToken).underlying()] = vToken;
```

Comment:
`vBNB` doesn't have the `underlying()` function. I would do an internal function `getUndelying` function, taking into account this special case and returning the WBNB address for the vBNB market (the ProtocolShareReserve will have WBNB, not BNB, so it makes sense).

When the user claim the boost associated to vBNB they will receive WBNB, that is acceptable